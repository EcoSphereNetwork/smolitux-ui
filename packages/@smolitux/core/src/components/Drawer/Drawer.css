.smolitux-drawer-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: var(--drawer-overlay-bg, rgba(0, 0, 0, 0.5));
  display: flex;
  z-index: var(--drawer-z-index, 1000);
  overflow: hidden;
}

.smolitux-drawer {
  position: fixed;
  background-color: var(--drawer-bg, white);
  display: flex;
  flex-direction: column;
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05);
  max-width: 100%;
  max-height: 100%;
  overflow: hidden;
}

/* Placement */
.smolitux-drawer--left {
  top: 0;
  left: 0;
  bottom: 0;
  width: var(--drawer-width, 320px);
  transform: translateX(-100%);
}

.smolitux-drawer--right {
  top: 0;
  right: 0;
  bottom: 0;
  width: var(--drawer-width, 320px);
  transform: translateX(100%);
}

.smolitux-drawer--top {
  top: 0;
  left: 0;
  right: 0;
  height: var(--drawer-height, 320px);
  transform: translateY(-100%);
}

.smolitux-drawer--bottom {
  bottom: 0;
  left: 0;
  right: 0;
  height: var(--drawer-height, 320px);
  transform: translateY(100%);
}

/* Open State */
.smolitux-drawer--open {
  transform: translate(0, 0);
}

/* Header */
.smolitux-drawer-header {
  display: flex;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid var(--drawer-border-color, #e5e7eb);
}

.smolitux-drawer-title {
  font-size: 1.125rem;
  font-weight: 600;
  color: var(--drawer-title-color, #111827);
  flex: 1;
}

.smolitux-drawer-close {
  background: none;
  border: none;
  cursor: pointer;
  color: var(--drawer-close-color, #6b7280);
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2rem;
  height: 2rem;
  border-radius: 9999px;
  transition: background-color 0.2s ease-in-out;
}

.smolitux-drawer-close:hover {
  background-color: var(--drawer-close-hover-bg, #f3f4f6);
  color: var(--drawer-close-hover-color, #1f2937);
}

.smolitux-drawer-close:focus {
  outline: none;
  box-shadow: 0 0 0 2px var(--primary-200, #bfdbfe);
}

/* Body */
.smolitux-drawer-body {
  flex: 1;
  overflow-y: auto;
  padding: 1rem;
  color: var(--drawer-body-color, #374151);
}

/* Footer */
.smolitux-drawer-footer {
  padding: 1rem;
  border-top: 1px solid var(--drawer-border-color, #e5e7eb);
}

/* Animations */
@keyframes smolitux-drawer-fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes smolitux-drawer-slide-in-left {
  from {
    transform: translateX(-100%);
  }
  to {
    transform: translateX(0);
  }
}

@keyframes smolitux-drawer-slide-in-right {
  from {
    transform: translateX(100%);
  }
  to {
    transform: translateX(0);
  }
}

@keyframes smolitux-drawer-slide-in-top {
  from {
    transform: translateY(-100%);
  }
  to {
    transform: translateY(0);
  }
}

@keyframes smolitux-drawer-slide-in-bottom {
  from {
    transform: translateY(100%);
  }
  to {
    transform: translateY(0);
  }
}

.smolitux-drawer-overlay--animated {
  animation: smolitux-drawer-fade-in 0.3s ease-out;
}

.smolitux-drawer--animated.smolitux-drawer--left.smolitux-drawer--open {
  animation: smolitux-drawer-slide-in-left 0.3s ease-out;
}

.smolitux-drawer--animated.smolitux-drawer--right.smolitux-drawer--open {
  animation: smolitux-drawer-slide-in-right 0.3s ease-out;
}

.smolitux-drawer--animated.smolitux-drawer--top.smolitux-drawer--open {
  animation: smolitux-drawer-slide-in-top 0.3s ease-out;
}

.smolitux-drawer--animated.smolitux-drawer--bottom.smolitux-drawer--open {
  animation: smolitux-drawer-slide-in-bottom 0.3s ease-out;
}

/* Responsive Adjustments */
@media (max-width: 640px) {
  .smolitux-drawer--left,
  .smolitux-drawer--right {
    width: 85vw;
  }
  
  .smolitux-drawer--top,
  .smolitux-drawer--bottom {
    height: 50vh;
  }
}

/* Dark Mode */
.dark .smolitux-drawer {
  background-color: var(--drawer-bg-dark, #1f2937);
}

.dark .smolitux-drawer-header {
  border-color: var(--drawer-border-color-dark, #374151);
}

.dark .smolitux-drawer-title {
  color: var(--drawer-title-color-dark, #f9fafb);
}

.dark .smolitux-drawer-close {
  color: var(--drawer-close-color-dark, #9ca3af);
}

.dark .smolitux-drawer-close:hover {
  background-color: var(--drawer-close-hover-bg-dark, #374151);
  color: var(--drawer-close-hover-color-dark, #f9fafb);
}

.dark .smolitux-drawer-body {
  color: var(--drawer-body-color-dark, #d1d5db);
}

.dark .smolitux-drawer-footer {
  border-color: var(--drawer-border-color-dark, #374151);
}