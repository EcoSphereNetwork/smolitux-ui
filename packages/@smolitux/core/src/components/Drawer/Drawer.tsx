// packages/@smolitux/core/src/components/Drawer/Drawer.tsx
import React, { useEffect, useRef } from 'react';

export type DrawerPlacement = 'left' | 'right' | 'top' | 'bottom';

export interface DrawerProps {
  /** Ist der Drawer sichtbar? */
  isOpen: boolean;
  /** Callback zum Schließen des Drawers */
  onClose: () => void;
  /** Position des Drawers */
  placement?: DrawerPlacement;
  /** Breite des Drawers (für left/right) */
  width?: string | number;
  /** Höhe des Drawers (für top/bottom) */
  height?: string | number;
  /** Titel des Drawers */
  title?: React.ReactNode;
  /** Footer-Inhalt */
  footer?: React.ReactNode;
  /** Inhalt des Drawers */
  children: React.ReactNode;
  /** Beim Klick auf Overlay schließen */
  closeOnOverlayClick?: boolean;
  /** Header-Anzeige */
  showHeader?: boolean;
  /** Zusätzliche CSS-Klassen */
  className?: string;
  /** Z-Index-Wert */
  zIndex?: number;
}

/**
 * Drawer-Komponente für Seitennavigationen und Panels
 * 
 * @example
 * ```tsx
 * <Drawer
 *   isOpen={isOpen}
 *   onClose={() => setIsOpen(false)}
 *   title="Seitenmenü"
 * >
 *   <p>Drawer-Inhalt</p>
 * </Drawer>
 * ```
 */
export const Drawer: React.FC<DrawerProps> = ({
  isOpen,
  onClose,
  placement = 'right',
  width = 320,
  height = 320,
  title,
  footer,
  children,
  closeOnOverlayClick = true,
  showHeader = true,
  className = '',
  zIndex = 50,
}) => {
  const drawerRef = useRef<HTMLDivElement>(null);
  
  // ESC-Taste zum Schließen
  useEffect(() => {
    const handleEscape = (e: KeyboardEvent) => {
      if (e.key === 'Escape' && isOpen) {
        onClose();
      }
    };
    
    document.addEventListener('keydown', handleEscape);
    return () => document.removeEventListener('keydown', handleEscape);
  }, [isOpen, onClose]);
  
  // Scroll des Body verhindern, wenn Drawer offen
  useEffect(() => {
    if (isOpen) {
      document.body.style.overflow = 'hidden';
    } else {
      document.body.style.overflow = '';
    }
    
    return () => {
      document.body.style.overflow = '';
    };
  }, [isOpen]);
  
  // Focus Trap
  useEffect(() => {
    if (isOpen && drawerRef.current) {
      const focusableElements = drawerRef.current.querySelectorAll(
        'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
      );
      
      if (focusableElements.length > 0) {
        (focusableElements[0] as HTMLElement).focus();
      }
    }
  }, [isOpen]);
  
  // Wenn nicht offen, nicht rendern
  if (!isOpen) return null;
  
  // Positionierungs-Styles
  const getPositionStyles = () => {
    const commonStyles = {
      position: 'fixed',
      zIndex: zIndex + 1,
      transition: 'all 0.3s ease-in-out',
    } as React.CSSProperties;
    
    switch (placement) {
      case 'left':
        return {
          ...commonStyles,
          left: 0,
          top: 0,
          bottom: 0,
          width: typeof width === 'number' ? `${width}px` : width,
          transform: isOpen ? 'translateX(0)' : 'translateX(-100%)',
          className: 'left-0',
        };
      case 'right':
        return {
          ...commonStyles,
          right: 0,
          top: 0,
          bottom: 0,
          width: typeof width === 'number' ? `${width}px` : width,
          transform: isOpen ? 'translateX(0)' : 'translateX(100%)',
          className: 'right-0',
        };
      case 'top':
        return {
          ...commonStyles,
          top: 0,
          left: 0,
          right: 0,
          height: typeof height === 'number' ? `${height}px` : height,
          transform: isOpen ? 'translateY(0)' : 'translateY(-100%)',
          className: 'top-0',
        };
      case 'bottom':
        return {
          ...commonStyles,
          bottom: 0,
          left: 0,
          right: 0,
          height: typeof height === 'number' ? `${height}px` : height,
          transform: isOpen ? 'translateY(0)' : 'translateY(100%)',
          className: 'bottom-0',
        };
      default:
        return commonStyles;
    }
  };
  
  // Horizontaler oder vertikaler Drawer?
  const isHorizontal = placement === 'left' || placement === 'right';
  
  return (
    <>
      {/* Overlay */}
      <div
        className="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
        style={{ zIndex }}
        onClick={closeOnOverlayClick ? onClose : undefined}
        aria-hidden="true"
        data-testid="drawer-overlay"
      />
      
      {/* Drawer */}
      <div
        ref={drawerRef}
        role="dialog"
        aria-modal="true"
        aria-labelledby={title ? 'drawer-title' : undefined}
        className={`
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          ${isHorizontal ? 'h-full' : 'w-full'}
          ${className}
          ${(getPositionStyles() as any).className || ''}
        `}
        style={getPositionStyles()}
      >
        {/* Header */}
        {showHeader && (
          <div className="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center">
            <h2 id="drawer-title" className="text-lg font-medium text-gray-900 dark:text-white">
              {title}
            </h2>
            <button
              type="button"
              className="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
              onClick={onClose}
              aria-label="Schließen"
            >
              <svg 
                className="h-6 w-6" 
                fill="none" 
                stroke="currentColor" 
                viewBox="0 0 24 24" 
                xmlns="http://www.w3.org/2000/svg"
              >
                <path 
                  strokeLinecap="round" 
                  strokeLinejoin="round" 
                  strokeWidth="2" 
                  d="M6 18L18 6M6 6l12 12"
                />
              </svg>
            </button>
          </div>
        )}
        
        {/* Content */}
        <div className="flex-grow overflow-y-auto p-4">
          {children}
        </div>
        
        {/* Footer */}
        {footer && (
          <div className="border-t border-gray-200 dark:border-gray-700 px-4 py-3">
            {footer}
          </div>
        )}
      </div>
    </>
  );
};

export default Drawer;
