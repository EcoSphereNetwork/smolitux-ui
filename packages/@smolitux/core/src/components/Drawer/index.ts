// packages/@smolitux/core/src/components/Drawer/index.ts
export { default, type DrawerProps, type DrawerPlacement } from './Drawer';
