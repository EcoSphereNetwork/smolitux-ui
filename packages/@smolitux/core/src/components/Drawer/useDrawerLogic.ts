import { useCallback, useEffect, useState } from 'react';

export type DrawerPlacement = 'left' | 'right' | 'top' | 'bottom';

export interface UseDrawerLogicOptions {
  defaultOpen?: boolean;
  placement?: DrawerPlacement;
  trapFocus?: boolean;
  closeOnEsc?: boolean;
  hasOverlay?: boolean;
}

export interface DrawerLogic {
  isOpen: boolean;
  placement: DrawerPlacement;
  trapFocus: boolean;
  hasOverlay: boolean;
  open: () => void;
  close: () => void;
  toggle: () => void;
}

export const useDrawerLogic = (
  options: UseDrawerLogicOptions = {}
): DrawerLogic => {
  const {
    defaultOpen = false,
    placement = 'right',
    trapFocus = true,
    closeOnEsc = true,
    hasOverlay = true,
  } = options;

  const [isOpen, setIsOpen] = useState(defaultOpen);

  const open = useCallback(() => setIsOpen(true), []);
  const close = useCallback(() => setIsOpen(false), []);
  const toggle = useCallback(() => setIsOpen((o) => !o), []);

  useEffect(() => {
    if (!closeOnEsc) return;
    const handle = (e: KeyboardEvent) => {
      if (e.key === 'Escape') {
        setIsOpen(false);
      }
    };
    window.addEventListener('keydown', handle);
    return () => window.removeEventListener('keydown', handle);
  }, [closeOnEsc]);

  return {
    isOpen,
    placement,
    trapFocus,
    hasOverlay,
    open,
    close,
    toggle,
  };
};
