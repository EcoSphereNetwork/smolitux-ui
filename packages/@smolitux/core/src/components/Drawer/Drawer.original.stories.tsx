import type { Meta, StoryObj } from '@storybook/react';
import { Drawer.original } from './Drawer.original';

const meta: Meta<typeof Drawer.original> = {
  title: 'Components/Drawer.original',
  component: Drawer.original,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Drawer.original',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Drawer.original',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Drawer.original',
    onClick: () => alert('Clicked!'),
  },
};
