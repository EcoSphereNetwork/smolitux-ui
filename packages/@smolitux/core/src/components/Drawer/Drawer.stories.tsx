import type { Meta, StoryObj } from '@storybook/react';
import { Drawer } from './Drawer';

const meta: Meta<typeof Drawer> = {
  title: 'Components/Drawer',
  component: Drawer,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Drawer',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Drawer',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Drawer',
    onClick: () => alert('Clicked!'),
  },
};
