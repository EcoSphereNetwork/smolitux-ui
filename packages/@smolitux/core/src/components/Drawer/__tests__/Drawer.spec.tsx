import React from 'react';
import { render } from '@testing-library/react';
import { Drawer } from '../Drawer';

describe('Drawer Snapshots', () => {
  const mockOnClose = jest.fn();

  it('renders default drawer correctly', () => {
    const { asFragment } = render(
      <Drawer isOpen={true} onClose={mockOnClose}>
        Drawer Content
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with title correctly', () => {
    const { asFragment } = render(
      <Drawer isOpen={true} onClose={mockOnClose} title="Drawer Title">
        Drawer Content
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with footer correctly', () => {
    const { asFragment } = render(
      <Drawer 
        isOpen={true} 
        onClose={mockOnClose} 
        footer={<button>Footer Button</button>}
      >
        Drawer Content
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with different placements correctly', () => {
    const placements: Array<'left' | 'right' | 'top' | 'bottom'> = ['left', 'right', 'top', 'bottom'];
    
    const fragments = placements.map(placement => {
      const { asFragment } = render(
        <Drawer isOpen={true} onClose={mockOnClose} placement={placement}>
          Drawer Content
        </Drawer>
      );
      return { placement, fragment: asFragment() };
    });
    
    fragments.forEach(({ placement, fragment }) => {
      expect(fragment).toMatchSnapshot(`Drawer with placement ${placement}`);
    });
  });

  it('renders with custom width correctly', () => {
    const { asFragment } = render(
      <Drawer isOpen={true} onClose={mockOnClose} placement="right" width="400px">
        Drawer Content
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom height correctly', () => {
    const { asFragment } = render(
      <Drawer isOpen={true} onClose={mockOnClose} placement="bottom" height="300px">
        Drawer Content
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without header correctly', () => {
    const { asFragment } = render(
      <Drawer 
        isOpen={true} 
        onClose={mockOnClose} 
        title="Drawer Title"
        showHeader={false}
      >
        Drawer Content
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom className correctly', () => {
    const { asFragment } = render(
      <Drawer isOpen={true} onClose={mockOnClose} className="custom-drawer">
        Drawer Content
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});