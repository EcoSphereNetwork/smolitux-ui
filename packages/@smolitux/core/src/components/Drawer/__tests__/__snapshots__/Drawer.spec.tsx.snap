// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Snapshots renders default drawer correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; right: 0px; top: 0px; bottom: 0px; width: 320px; transform: translateX(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with custom className correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          custom-drawer
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; right: 0px; top: 0px; bottom: 0px; width: 320px; transform: translateX(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with custom height correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          w-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; bottom: 0px; left: 0px; right: 0px; height: 300px; transform: translateY(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with custom width correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; right: 0px; top: 0px; bottom: 0px; width: 400px; transform: translateX(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with different placements correctly: Drawer with placement bottom 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          w-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; bottom: 0px; left: 0px; right: 0px; height: 320px; transform: translateY(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with different placements correctly: Drawer with placement left 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; left: 0px; top: 0px; bottom: 0px; width: 320px; transform: translateX(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with different placements correctly: Drawer with placement right 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; right: 0px; top: 0px; bottom: 0px; width: 320px; transform: translateX(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with different placements correctly: Drawer with placement top 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          w-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; top: 0px; left: 0px; right: 0px; height: 320px; transform: translateY(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with footer correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; right: 0px; top: 0px; bottom: 0px; width: 320px; transform: translateX(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      />
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
    <div
      class="border-t border-gray-200 dark:border-gray-700 px-4 py-3"
    >
      <button>
        Footer Button
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders with title correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-labelledby="drawer-title"
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; right: 0px; top: 0px; bottom: 0px; width: 320px; transform: translateX(0);"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-700 px-4 py-3 flex justify-between items-center"
    >
      <h2
        class="text-lg font-medium text-gray-900 dark:text-white"
        id="drawer-title"
      >
        Drawer Title
      </h2>
      <button
        aria-label="Close"
        class="text-gray-400 hover:text-gray-500 dark:text-gray-500 dark:hover:text-gray-400"
        type="button"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Drawer Snapshots renders without header correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-black bg-opacity-50 transition-opacity duration-300"
    style="z-index: 50;"
  />
  <div
    aria-labelledby="drawer-title"
    aria-modal="true"
    class="
          bg-white dark:bg-gray-800 shadow-xl flex flex-col
          h-full
          
        "
    role="dialog"
    style="position: fixed; z-index: 51; transition: all 0.3s ease-in-out; right: 0px; top: 0px; bottom: 0px; width: 320px; transform: translateX(0);"
  >
    <div
      class="flex-grow overflow-y-auto p-4"
    >
      Drawer Content
    </div>
  </div>
</DocumentFragment>
`;
